src/components/HeroCard/HeroCard.js

```diff
@@ -4,21 +4,40 @@ import Column from '../Column';
 import HeroIcon from '../HeroIcon';
 import CounterCard from '../CounterCard';
 import CounterTabs from '../CounterTabs';
-import HeroData from "../../utils/heroes.json";
+
+
+
+const getCounter = (heroData, counter) => {
+
+        for(let key in heroData) {
+            if (counter.id === Number(key)) {
+                return heroData[key];
+            }
+        }
+}
 
 export default class HeroCard extends Component {
     state = {
         "hero": null,
         "tabState": false,
-        "currentTab": null
+        "currentTab": null,
+        "counterOne": null,
+        "counterTwo": null,
+        "counterThree": null,
+        "counterInfo": null
     }
 
+    
 
     componentWillMount() {
 
+        const {hero} = this.props;
         this.setState({
-            "hero": HeroData
-        })
+            "hero": hero,
+            counterOne: getCounter(this.props.heroData, hero.counters[0]),
+            counterTwo: getCounter(this.props.heroData, hero.counters[1]),
+            counterThree: getCounter(this.props.heroData, hero.counters[2]),
```

Comment:
Consider that using this technique does not allow you to easily iterate over the counters. What if you used an array, thusly:

```javascript
const {heroData} = this.props;
const {counters} = hero;

this.setState({
    hero,
    counters: counters.map(({id}) => {
        return heroData[id];
    })
});
```

Reply:
Interesting... I'm not mapping the CounterTabs, which I should be. I will have to look at this more heavily.